### DISCUSSION TITLE
Extracting .gds files from the adjoint solver

### DISCUSSION THREAD

**Discussion by mijna37** [NanoComp/meep#2620] [Q&A] [answered]
Is there a standard way in MEEP to export the gds files or a recommended library to do so? 

I have the .png and .csv files for my final topology design with values ranging between 0 to 1. I think the way to go is to set a specific threshold value that determines whether each pixel is void or not, and afterwards use gdsfactory to set this pixel to i.e. silicon or air and assemble the gds. Further, is it wise to increase design_region_resolution so that the final gds is as smooth as possible?

**Comment by smartalecH** [comment, marked as answer]
There's no standard approach endorsed by meep, but there are many python image processing libraries that can extract contours from an image (eg your design). Those contours will each have vertices that can be exported to gds shapes using something like gdspy. This is the approach we typically take.
